src/compiler/checker.ts

```diff
@@ -14597,6 +14599,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
         }
         const props = propSet ? arrayFrom(propSet.values()) : [singleProp];
+        const id = `${checkFlags & CheckFlags.ReadPartial ? 'p' : ''}${skipObjectFunctionPropertyAugment ? 's' : ''}${isUnion ? 'u' : 'i'}${getSymbolListId(props)}`;

```

Comment:
This id is a little bit ad-hoc so it might be missing some important. To the best of my understanding (and that might not be complete here) this sounds reasonably well - unless there is just some fundamental reason why this is blatantly wrong that I missed.